packages/next/src/server/lib/incremental-cache/file-system-cache.ts

```diff
@@ -323,6 +313,17 @@ export default class FileSystemCache implements CacheHandler {
             console.log('FileSystemCache: expired tags', cacheTags)
           }
 
+          // If allowStale is enabled, return stale data with lastModified = -1
+          // to trigger background revalidation instead of blocking
+          if ('allowStale' in ctx && ctx.allowStale) {
+            if (FileSystemCache.debug) {
+              console.log(
+                'FileSystemCache: serving stale for expired tags with allowStale'
+              )
+            }
+            return { ...data, lastModified: -1 }
```

Comment:
tweaked this to return `lastModified: -1` rather than the stale data so it will still do a background revalidation